src/middlewared/middlewared/plugins/failover_/detect_enclosure.py

```diff
@@ -22,10 +22,21 @@ class Config:
     @cache
     def detect(self):
         HARDWARE = NODE = 'MANUAL'
-        product = self.middleware.call_sync('system.dmidecode_info')['system-product-name']
+        dmi = self.middleware.call_sync('system.dmidecode_info')
+        product = dmi['system-product-name']
         if not product:
             # no reason to continue since we've got no path forward
             return HARDWARE, NODE
+        elif dmi['system-manufacturer'] == 'QEMU':
+            serial = dmi['system-serial-number']
+            if not serial.startswith('ha') and not serial.endswith(('_c1', '_c2')):
+                # truenas is often installed in KVM so we need to check our specific
+                # strings in DMI and bail out early here
+                return HARDWARE, NODE
```

Comment:
Should we have a persistent alert that this is unsupported configuration for production workloads in case someone sees the source and decides to write a howto?